bio/quast/wrapper.py

```
@@ -67,9 +71,62 @@
+    ### Copy files to final destination
+    def save_output(src, dst, wd=Path(".")):
+        if not dst:
+            return 0
+        dest = wd / dst
+        shell("cat {src} > {dest}")
 
-shell(
-    "quast --threads {snakemake.threads} {ref} {gff} {pe1} {pe2} {pe12} {mp1} {mp2} {mp12} {single} {pacbio} {nanopore} {ref_bam} {ref_sam} {bam} {sam} {sv_bedpe} {extra} -o {output_dir} {snakemake.input.fasta} {log}"
-)
+    ### Saving LOG files
+    save_output(f"{tmpdir}/quast.log", snakemake.log.get("quast"))
```

Ruff

Undefined name `snakemake`

(F821)